File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonNetworking/src/main/java/triathlon/networking/protocol/dto/DTOUtils.java
package triathlon.networking.protocol.dto;

import triathlon.model.Participant;
import triathlon.model.Result;
import triathlon.model.Trial;

public class DTOUtils {
    public static Result getFromDTO(ResultDTO resultDTO) {
        Participant participant = new Participant(null, null);
        participant.setId(resultDTO.getIdParticipant());
        Trial trial = new Trial(null, null, null);
        trial.setId(resultDTO.getIdTrial());
        return new Result(participant, trial, resultDTO.getPoints());
    }

    public static ResultDTO getDTO(Result result) {
        Long idParticipant = result.getParticipant().getId();
        Long idTrial = result.getTrial().getId();
        return new ResultDTO(idParticipant, idTrial, result.getPoints());
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/networking/dtos/DTOUtils.cs
using model;

namespace networking.dtos;

public class DTOUtils
{
    public static Result GetFromDTO(ResultDTO resultDto)
    {
        Participant participant = new Participant("", "");
        participant.Id = resultDto.IdParticipant;
        Trial trial = new Trial("", "", null!);
        trial.Id = resultDto.IdTrial;
        return new Result(participant, trial, resultDto.Points);
    }

    public static ResultDTO GetDTO(Result result)
    {
        long idParticipant = result.Participant.Id;
        long idTrial = result.Trial.Id;
        return new ResultDTO(idParticipant, idTrial, result.Points);
    }
}